crates/ruff_source_file/src/lib.rs

```diff
@@ -253,3 +253,20 @@ impl Debug for SourceLocation {
             .finish()
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum SourceRow {
+    /// A row within a cell in a Jupyter Notebook.
+    Notebook(OneIndexed, OneIndexed),
```

Comment:
nit: I'd give those explicit names given that the description mentions first row then cell while the data is first cell then row.